image/docker_schema2.go

```diff
@@ -173,6 +168,50 @@ func (m *manifestSchema2) UpdatedImage(options types.ManifestUpdateOptions) (typ
 	return memoryImageFromManifest(&copy), nil
 }
 
+// Edits the layers in the manifest when called by replacing them with the
+// appropriate infos provided. This is metadata-only -- the actual layers still
+// have to get to the image.
+func (m *manifestSchema2) performEdits(infos []types.BlobInfo) error {
+	infolen := len(infos)
+
+	if infolen < 1 {
+		return errors.New("cannot edit an image down to less than one layer")
+	}
+
+	// regurgitate the image configuration for recalculation of layers in the event the layer list has been edited.
+	imageConfig := &image{}
```

Comment:
There is a fairly high risk that this would drop any fields of `config.json` added in the future.